Automattic/sensei
# Add student submit quiz email, refactor email classes, separate pattern structures

Resolves https://github.com/Automattic/sensei/issues/6481

## Proposed Changes
* Added email for informing the teacher when a student submits a quiz
* Refactor - Move email patterns to separate files
* Refactor - Separate email sending classes for different events
* Refactor - Handle disabled email centrally

## Testing Instructions

- Enable the email post type
- Do not forget to run the tool (Sensei LMS -> Tools -> Re Create Emails). Run this again even if you've ran it before
- Create a course
- Have a lesson in that course with a quiz that will need to be manually graded (Email is sent to the teacher only when the quiz needs to be manually graded, quizzes that are auto-gradeable do not trigger email)
- Take the quiz as a student
- Make sure the teacher received an email

## Screenshot

![image](https://user-images.githubusercontent.com/6820724/221899835-0ab3ca85-16ff-4365-95c2-d20acc494b0c.png)

## Hooks

### Added

**sensei_email_generators** -- Allows modifying the email classes that are being registered

## Pre-Merge Checklist
